### DISCUSSION TITLE
beforeInteractive Script in import

### DISCUSSION THREAD

**Discussion by ysulyma** [vercel/next.js#46009] [Help] [answered]
I have some `<Script>`s using `strategy="beforeInteractive"`. To keep my code organized, I'd like to separate them into their own files, rather than inline in `_document.tsx`. My code looks like this:
```tsx
// pages/_document.tsx
import {Html, Head, Main, NextScript} from "next/document";

import {ThirdPartyScript} from "../components/tags/ThirdParty";

export default function Document() {
  return (
    <Html>
      <Head>
        <link rel="shortcut icon" href="/favicon.ico" />
        <ThirdPartyScript />
      </Head>
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  );
}

// components/tags/ThirdParty.tsx
import Script from "next/script";

export function ThirdPartyScript() {
  return (
    <Script strategy="beforeInteractive">
      {/* long inline script */}
    </Script>
  );
}
```

This is giving me a warning

> `next/script`'s `beforeInteractive` strategy should not be used outside of `pages/_document.js`. See: [@next/next/no-before-interactive-script-outside-document](https://nextjs.org/docs/messages/no-before-interactive-script-outside-document).

Is it safe to disable the lint warning in this case, since it actually is getting rendered into `_document.tsx`? Or is Next doing some magic where it needs to directly appear in `_document.tsx`, not imported?

**Comment by guitheengineer** [comment, marked as answer]
It's safe to use that way, the warning is probably a bug in this case because it's just verifying the filename where it's used, not where the component it's actually is being called. This could be an issue.
